llvm/test/CodeGen/X86/ctselect-constant-time.ll

```diff
@@ -0,0 +1,159 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -mtriple=i386-unknown-linux-gnu -mattr=-cmov | FileCheck %s --check-prefix=CONSTANT-TIME
+; RUN: llc < %s -mtriple=i386-unknown-linux-gnu -mattr=-cmov -verify-machineinstrs | FileCheck %s --check-prefix=CONSTANT-TIME
+
+; Test constant-time properties of CTSELECT implementation on i386
+; This test verifies that:
+; 1. No conditional branches are generated
+; 2. Both true and false values are always loaded/computed
+; 3. No conditional memory loads are performed
+
+define i32 @crypto_key_select(i32 %secret_bit, i32 %key1, i32 %key2) nounwind {
+; CONSTANT-TIME-LABEL: crypto_key_select:
+; CONSTANT-TIME:       # %bb.0:
+; CONSTANT-TIME-NEXT:    pushl %ebx
+; CONSTANT-TIME-NEXT:    pushl %edi
+; CONSTANT-TIME-NEXT:    pushl %esi
+; CONSTANT-TIME-NEXT:    movl {{[0-9]+}}(%esp), %ecx
+; CONSTANT-TIME-NEXT:    movl {{[0-9]+}}(%esp), %edx
+; CONSTANT-TIME-NEXT:    cmpl $0, {{[0-9]+}}(%esp)
+; CONSTANT-TIME-NEXT:    setne %al
+; CONSTANT-TIME-NEXT:    testb %al, %al
+; CONSTANT-TIME-NEXT:    BUNDLE
```

Comment:
It seems unit test is only generating `BUNDLE` header opcode. This could be one reason of failing tests.